src/Voron/TransactionPersistentContext.cs

```diff
@@ -32,20 +32,19 @@ public PageLocator AllocatePageLocator(LowLevelTransaction tx)
             if (_pageLocators.Count != 0)
             {
                 locator = _pageLocators.Pop();
-                locator.Renew(tx);
+                locator.Renew();
             }
             else
             {
-                locator = new PageLocator(tx);
+                locator = new PageLocator();
             }
             return locator;
         }
 
         public void FreePageLocator(PageLocator locator)
         {
             Debug.Assert(locator != null);
-            locator.Release();
-            if (_pageLocators.Count < 1024)
+            if (_pageLocators.Count < 256)
```

Comment:
What is the reasoning for limiting it to 256 instead of 1024?

Reply:
The extra memory consumption. Reverted now that I was able to keep it in 20 bytes.